### ISSUE TITLE
Environment variables are not recognized in task name

### ISSUE THREAD

**Issue by mclate** [ansible/ansible#12280]
##### Issue Type:

Bug report
##### Ansible Version:

```
ansible 1.9.3
  configured module search path = /usr/share/ansible
```
##### Ansible Configuration:

No extra changes applied
##### Environment:

N/A
##### Summary:

In role, when variable is substituted into task name, if one is taken from role invocation (like `{role: role, variable: value}`), it is substituted correctly into task name (`- name: Task {{variable}}` results in `Task value`), however, if variable is not provided in role definition but provided in any other way (such as playbook invocation argument), it is not recognized: mentioned above task name results in `Task {{variable}}`. However, variable itself is recognized and substituted correctly in task details (see "Steps To Reproduce" for details)
##### Steps To Reproduce:
1. Create playbook:

``` yaml
- hosts: all
- roles:
    - role: my_role
```
1. Create role `my_role`:

``` yaml
- name: Debug {{ variable1 }}
  debug:
      var: "{{ variable1 }}"
```
1. Run playbook as `ansible-playbook playbook.yml -e variable1=value1`
##### Expected Results:

``` ansible
TASK: [my_role | Debug value1] ************************************************
ok: [host1] => {
    "var": {
        "variable1": "value1"
    }
}
```
##### Actual Results:

```
TASK: [my_role | Debug {{ variable1 }}] ************************************************
ok: [host1] => {
    "var": {
        "variable1": "value1"
    }
}
```

As you can see, variable is correctly substituted in debug statement, but is not recognized in task name.

However, if we replace role invocation as:

``` yaml
- hosts: all
- roles:
    - role: my_role
      variable1: "{{ variable1 | default('default') }}"
```

We receive expected result. (`default('default')` is used because in case if variable is not defined we will receive recurcive loop when templating the variable)

Also, if we replace task name as (without `variable1` definition in role invocation):

``` yaml
- name: Debug {{ variable1 | default('default') }}
  debug:
      var: "{{ variable1 }}"
```

we get 

``` ansible
TASK: [my_role | Debug default] ************************************************
ok: [host1] => {
    "var": {
        "variable1": "value1"
    }
}
```

**Comment by bcoca** [comment]
in 1.9 templating names was pretty much broken, can you test against current devel?

**Comment by mclate** [comment]
Can confirm it works in ansible `2.0.0`

```
ansible 2.0.0 (devel 246e602fce) last updated 2015/09/08 14:21:26 (GMT +000)
  lib/ansible/modules/core: (detached HEAD 34655e8e29) last updated 2015/09/08 14:21:38 (GMT +000)
  lib/ansible/modules/extras: (detached HEAD 6a3cf63351) last updated 2015/09/08 14:21:48 (GMT +000)
  config file =
  configured module search path = None
```

`ansible-playbook playbook.yml -e var=val`:

```
TASK [my_role : Debug val] *****************************************************
ok: [localhost] => {
    "changed": false,
    "{{ var }}": "val"
}
```

**Comment by bcoca** [comment]
as per our policy I'm going to close this ticket as it is fixed in devel and will be available in upcoming 2.0 release. Due to the major changes done in core to enable this I doubt we can backport to the 1.9 tree.

**Comment by bcoca** [comment]
# Closing This Ticket

Hi!

We believe recent commits (likely detailed above) should resolve this question or problem for you.

This will also be included in the next major release.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular
issue is resolved.

Thank you!
